ports/espressif/common-hal/wifi/Radio.c

```diff
@@ -376,7 +376,10 @@ wifi_radio_error_t common_hal_wifi_radio_connect(wifi_radio_obj_t *self, uint8_t
     } while ((bits & (WIFI_CONNECTED_BIT | WIFI_DISCONNECTED_BIT)) == 0 && !mp_hal_is_interrupted());
 
     if ((bits & WIFI_DISCONNECTED_BIT) != 0) {
-        if (self->last_disconnect_reason == WIFI_REASON_AUTH_FAIL) {
+        if (
+            (self->last_disconnect_reason == WIFI_REASON_AUTH_FAIL) ||
+            (self->last_disconnect_reason == WIFI_REASON_4WAY_HANDSHAKE_TIMEOUT)
+            ) {
             return WIFI_RADIO_ERROR_AUTH_FAIL;
         } else if (self->last_disconnect_reason == WIFI_REASON_NO_AP_FOUND) {
             return WIFI_RADIO_ERROR_NO_AP_FOUND;
```

Comment:
I agree with @anecdata's suggestion. Just delete these lines.
```suggestion
```

Reply:
This isn't correct because `WIFI_**REASON**_NO_AP_FOUND` may not be the same thing as `WIFI_**RADIO**_ERROR_NO_AP_FOUND`. This is translating from ESP-IDF errors to circuitpython ones. I'm not sure why we return the reason otherwise. Maybe to at least show some number.

Reply:
Please undo the last commit with this.

Reply:
Reverted.

Reply:
Whoops, sorry!

Reply:
Sorry, I missed that, thought they looked the same.